docs/content/docs/tutorials/emr-on-eks-example.md

```diff
@@ -0,0 +1,494 @@
+    Type:                  ACK.ResourceSynced
+  Id:                      dxnqujbxexzri28ph1wspbxo0
+Events:                    <none>
+```
+
+## Run a Sample Spark Job
+In order to run sample spark job, we need to create EMR Job Execution Role. This Role will have IAM permissions such as S3, CloudWatch Logs for running your job. We will use IRSA to secure this job role
+```
+ACK_JOB_EXECUTION_ROLE="ack-${SERVICE}-jobexecution-role"
+ACK_JOB_EXECUTION_IAM_ROLE_DESCRIPTION="IRSA role for ACK ${SERVICE} Job Execution"
+
+cat <<EOF > job_trust.json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "ec2.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+EOF
+aws iam create-role --role-name "${ACK_JOB_EXECUTION_ROLE}" \
+    --assume-role-policy-document file://job_trust.json  \
+    --description "${ACK_JOB_EXECUTION_IAM_ROLE_DESCRIPTION}"
+
+export ACK_JOB_EXECUTION_ROLE_ARN=$(aws iam get-role --role-name=$ACK_JOB_EXECUTION_ROLE --query Role.Arn --output text)    
+```
+```
+cat <<EOF > job_policy.json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:PutObject",
+                "s3:GetObject",
+                "s3:ListBucket"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:PutLogEvents",
+                "logs:CreateLogStream",
+                "logs:DescribeLogGroups",
+                "logs:DescribeLogStreams"
+            ],
+            "Resource": [
+                "arn:aws:logs:::"
+            ]
+        }
+    ]
+}
+EOF
+echo "Creating ACK-${SERVICE}-JobExecution-POLICY"
+aws iam create-policy   \
+  --policy-name ack-${SERVICE}-jobexecution-policy \
+  --policy-document file://job_policy.json
+
+echo -n "Attaching IAM policy ..."
+aws iam attach-role-policy \
+  --role-name "${ACK_JOB_EXECUTION_ROLE}" \
+  --policy-arn "arn:aws:iam::${AWS_ACCOUNT_ID}:policy/ack-${SERVICE}-jobexecution-policy"
+
+aws emr-containers update-role-trust-policy \
+  --cluster-name ${EKS_CLUSTER_NAME} \
+  --namespace ${EMR_NAMESPACE} \
+  --role-name ${ACK_JOB_EXECUTION_ROLE}
+```
+Now let's submit sample spark job
+```
+echo "checking if VirtualCluster Status is "True""
+VC=$(kubectl get virtualcluster -o jsonpath='{.items..metadata.name}')
+kubectl describe virtualcluster/$VC | yq e '.Status.Conditions.Status'
+
+echo "get VirtualCluster ID"
+export VC_ID=$(kubectl describe virtualcluster/$VC | yq e '.Status.Id')
+
+export RANDOM_ID=$(LC_ALL=C tr -dc a-z0-9 </dev/urandom | head -c 8)
+
+cat << EOF > jobrun.yaml
+---
+apiVersion: emrcontainers.services.k8s.aws/v1alpha1
+kind: JobRun
+metadata:
+  name: my-ack-jobrun-${RANDOM_ID}
+spec:
+  name: my-ack-jobrun-${RANDOM_ID}
+  virtualClusterID: "${VC_ID}"
```

Comment:
Let's use resource references here! Let's add support for references on this field (and any others)

Reply:
I am not sure what you mean. Can you add more details of what you are looking for?

Reply:
Here is the design doc for resource references with ACK: https://github.com/aws-controllers-k8s/community/blob/main/docs/design/proposals/resource-reference/resource-reference.md

Essentially, you tell the code generator that a given field is a reference to a property of another ACK resource. In this case, `virtualClusterID` should be the ID from a `VirtualCluster` resource. Then, users can create a `VirtualCluster` ACK resource, and use the name of the K8s object within `JobRun`, and the controller will look up that resource's ID and inject it into the spec. Therefore you don't need to export and import any values manually, or if the `VirtualCluster` changes ID the spec is automatically updated.

Reply:
fixed in new commit